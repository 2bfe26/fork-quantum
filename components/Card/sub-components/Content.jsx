import styled from 'styled-components';

const Content = styled.div`
  padding: 12px 20px 20px;
  font-size: 14px;
  margin: 0;
`;

Content.displayName = 'Card.Content';

export default Content;
