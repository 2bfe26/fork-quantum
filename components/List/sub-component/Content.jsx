import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Header from './Header';
import SubHeader from './SubHeader';

const ContentBox = styled.div`
  display: flex;
  flex-direction: column;
`;

class Content extends React.Component {
  _renderContent = (content) => {
    if (typeof content === 'string') {
      return <Header>{content}</Header>;
    }

    const {
      header = this.props.header,
      subheader = this.props.subheader,
    } = content;

    return (
      <React.Fragment>
        <Header>{header}</Header>

        {subheader && <SubHeader>{subheader}</SubHeader>}
      </React.Fragment>
    );
  };

  render() {
    const { content, children } = this.props;

    return (
      <ContentBox>
        {children || this._renderContent(content)}
      </ContentBox>
    );
  }
}

Content.defaultProps = {
  subheader: '',
  header: '',
  content: {},
  children: null,
};

Content.propTypes = {
  header: PropTypes.string,
  subheader: PropTypes.string,
  content: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      header: PropTypes.string,
      subheader: PropTypes.string,
    }),
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Content;
