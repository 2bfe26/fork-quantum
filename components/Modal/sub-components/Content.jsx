import styled from 'styled-components';
import Card from '../../Card';

const Content = styled(Card.Content)`
  font-size: 16px;
  max-height: 70vh;
  overflow-y: auto;
  padding: 16px 24px;
`;

export default Content;
