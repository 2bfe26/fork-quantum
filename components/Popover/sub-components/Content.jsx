import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import Colors from '../../Colors';
import { components, spacing } from '../../shared/theme';
import getArrow from '../arrowProperties';
import Button from '../../Button';

const _colors = ({
  skin,
  theme: {
    components: {
      popover: {
        skins: { [skin]: popoverColor },
      },
    },
  },
  inverted,
}) => {
  if (inverted) {
    const [background, text] = [popoverColor.text, popoverColor.background];

    return css`
      background-color: ${background};
      color: ${text};
    `;
  }
  return css`
    background-color: ${popoverColor.background};
    color: ${popoverColor.text};
  `;
};

const PopoverContent = styled.div`
  box-shadow: 0 2px 4px 0 ${Colors.SHADOW[50]};
  align-items: start;
  display: flex;
  border-radius: 4px;
  font-size: 16px;
  ${({
    theme: {
      spacing: { xsmall },
    },
  }) => `padding: ${xsmall}px`};
  position: absolute;
  line-height: 0;
  transition: opacity 0.2s ease-in-out, visibility 0.2s ease-in-out;
  z-index: 100;

  &:before {
    ${({ placement, skin, theme, inverted }) =>
      getArrow(placement, skin, theme, inverted)};
  }

  ${_colors}
`;

const CloseButton = styled(Button.Icon).attrs({
  icon: 'close',
})`
  display: inherit;
  height: auto;
  ${({
    theme: {
      spacing: { medium },
    },
  }) => `margin-left: ${medium}px`};
  opacity: 0.8;
  padding: 0;
  transition: opacity 0.4s ease;
  width: auto;

  :hover {
    background: none;
    opacity: 1;
  }
`;

const PopoverChildren = styled.div``;

class Content extends Component {
  constructor(props) {
    super(props);

    this.wrapper = document.createElement('section');
  }

  componentDidMount() {
    document.body.appendChild(this.wrapper);
  }

  componentWillUnmount() {
    document.body.removeChild(this.wrapper);
  }

  render() {
    const { placement, children, onPopoverClose, theme, ...rest } = this.props;

    return ReactDOM.createPortal(
      <PopoverContent
        theme={theme}
        placement={placement}
        ref={element => {
          this.innerContentRef = element;
        }}
        {...rest}
      >
        <PopoverChildren>{children}</PopoverChildren>
        <CloseButton theme={theme} onClick={onPopoverClose} />
      </PopoverContent>,
      this.wrapper,
    );
  }
}

CloseButton.displayName = 'CloseButton';
PopoverChildren.displayName = 'PopoverChildren';

Content.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  inverted: PropTypes.bool,
  placement: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
  onPopoverClose: PropTypes.func,
  skin: PropTypes.oneOf(['neutral', 'success', 'warning', 'error']),
  theme: PropTypes.shape({
    spacing: PropTypes.object,
    components: PropTypes.shape({
      badge: PropTypes.object,
    }),
  }),
};

Content.defaultProps = {
  inverted: false,
  placement: 'top',
  onPopoverClose: () => {},
  skin: 'neutral',
  theme: {
    spacing,
    components: {
      popover: components.popover,
    },
  },
};

export default Content;
