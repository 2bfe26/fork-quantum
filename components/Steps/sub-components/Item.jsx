import React from 'react';
import PropTypes from 'prop-types';

const Item = ({title, active, icon}) => {
  console.log(active);

  const isActive = active ? 'active' : '';

  return (
    <React.Fragment>
      <li className={isActive}>
        <p>{title}</p>
        <div style={{position: 'relative', top: -60}}>s</div>
      </li>
    </React.Fragment>
  );
};

Item.propTypes = {
  title: PropTypes.string,
  icon: PropTypes.string,
  active: PropTypes.bool
};

Item.defaultProps = {
  active: false
};

export default Item;
