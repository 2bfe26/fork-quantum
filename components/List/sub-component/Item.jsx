import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Icon from '../../Icon';
import Content from './Content';

const ItemIcon = styled(Icon)`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 8px;
`;

const bulletValue = ({ bullet }) =>
  bullet &&
  css`
    &:before {
      content: '${bullet}';
    }
  `;

const ListItem = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  padding: 4px 8px 8px 4px;

  margin-bottom: 4px;

  ${bulletValue};
`;

class Item extends React.Component {
  static create = (item) => {
    if (typeof item === 'string') {
      return <Item content={item} key={item} />;
    }

    return <Item {...item} key={item.content} />;
  };

  _renderIcon = (icon) => {
    if (typeof icon === 'string') {
      return <ItemIcon name={icon} />;
    }

    return <ItemIcon {...icon} />;
  }

  render() {
    const {
      icon,
      children,
      content,
      bullet,
      ...rest
    } = this.props;

    return (
      <li {...rest}>
        <ListItem bullet={bullet} >
          {icon && this._renderIcon(icon)}
          {children || <Content content={content} />}
        </ListItem>
      </li>
    );
  }
}

Item.defaultProps = {
  icon: '',
  bullet: '',
  children: null,
};

Item.propTypes = {
  content: PropTypes.oneOf(
    PropTypes.string,
    PropTypes.shape({
      subheader: PropTypes.string.isRequired,
      header: PropTypes.string,
    }),
  ).isRequired,
  icon: PropTypes.string,
  bullet: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Item;
