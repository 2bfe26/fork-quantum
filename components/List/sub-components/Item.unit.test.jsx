import React from 'react';
import { shallow } from 'enzyme';
import Item from './Item';
import Content from './Content';

const content = 'content';
const icon = 'icon';
const header = 'header';
const subheader = 'subheader';

describe('<Item />', () => {
  it('Should match the snapshot', () => {
    expect(shallow(<Item />)).toMatchSnapshot();
  });

  it('render an Item with icon', () => {
    const wrapper = shallow(<Item icon={icon} />);

    expect(wrapper.find('ItemIcon')).toHaveLength(1);
  });

  it('render Content component', () => {
    const wrapper = shallow(<Item content={content} />);

    expect(wrapper.render().text()).toBe(content);
    expect(wrapper.find(Content)).toHaveLength(1);
  });

  it('accept any children as a content', () => {
    const wrapper = shallow(<Item>{content}</Item>);

    expect(wrapper.render().text()).toBe(content);
    expect(wrapper.find(Content)).toHaveLength(0);
  });

  it('render Content, Header and SubHeader components', () => {
    const wrapper = shallow(<Item content={{ header, subheader }} />);

    expect(wrapper.find(Content)).toHaveLength(1);
    expect(wrapper.render().text()).toContain(header);
    expect(wrapper.render().text()).toContain(subheader);
  });
});
