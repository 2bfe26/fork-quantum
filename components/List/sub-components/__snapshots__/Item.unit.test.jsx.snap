// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List.Item
    bullet=""
    content={null}
    icon=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ListItem
        bullet=""
      >
        
        <Content
          content={null}
          header=""
          subheader=""
        />
      </ListItem>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bullet": "",
        "children": Array [
          "",
          <Content
            content={null}
            header=""
            subheader=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
            "content": null,
            "header": "",
            "subheader": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ListItem
          bullet=""
        >
          
          <Content
            content={null}
            header=""
            subheader=""
          />
        </ListItem>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bullet": "",
          "children": Array [
            "",
            <Content
              content={null}
              header=""
              subheader=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": null,
              "content": null,
              "header": "",
              "subheader": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
